WordPress/gutenberg
# Gallery block: Add a dismissible warning about the need for versions >=18.2 of the mobile app

## Description
With the new Gallery block there is the potential for data loss if a user adds a Gallery block via web, and then edits in on the mobile app with a version < 18.2. There are currently around 12% of users on older versions and now way to prevent these users editing the new block format, or to notify them via the app about this.

This PR adds a dismissible snackbar notice when a new gallery block is added for the first time in the web.  It is only a temporary notice and can be removed some time before the 6.0 release.

## How has this been tested?

- Check out this PR and enable the gallery experiment
- Add a gallery block
- Check that the mobile warning snackbar displays
- Dismiss the notice and make sure that it doesn't display again when adding a new gallery, even after a browser restart
- Clear local storage for test site to retest

## Screenshots

![snackbar](https://user-images.githubusercontent.com/3629020/140422175-7cd37b96-d943-43c8-be60-9f8e5fa20ba0.gif)
